.changeset/curly-jeans-fly.md

```diff
@@ -0,0 +1,5 @@
+---
+"@biomejs/biome": patch
+---
+
+Added the new nursery rule [`noVueArrowFuncInWatch`](https://biomejs.dev/linter/rules/no-vue-arrow-func-in-watch/). This rule forbids using arrow functions in watchers in Vue components, because in an arrow function `this` won't reference the Vue instance.
```

Comment:
I don't really understand the explanation of the rule. `this` is only available in normal functions...

Reply:
It's pretty Vue specific. The rule is a direct port of https://eslint.vuejs.org/rules/no-arrow-functions-in-watch.html

In a Vue component (options api), `this` is supposed to refer to "this instance of this component". Vue calls `.bind()` on these functions to set `this`, but (i think) it doesn't work on arrow functions because an arrow function immediately binds itself to the parent context.

Reply:
I understand, however if people use arrow functions, _there's no `this` to reference_ from, and I think that's the whole point of the rule. If you need access to the component instance, you must use classic functions. 

The way it is phrased, it seems there's a `this` somewhere, however there isn't.

I would just say that arrow functions don't allow access to the component instance, while classic functions do via `this`.